src/debugger/message/MessageQueue.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import { Message } from "./Message.js";
+
+export class MessageQueue {
+  constructor(fs: any, outFilePath: string) {
+    this._queue = [];
+    this._awaitingResponse = false;
+
+    this._fs = fs;
+    this._outFilePath = outFilePath;
+  }
+  _queue: Array<Message>;
+  _awaitingResponse: boolean;
+  _fs: any;
+  _outFilePath: string;
+
+  enqueue(message: Message) {
+    this._queue.push(message);
+  }
+
+  dequeue(): Message | void {
+    if (this._queue.length === 0) {
+      return undefined;
```

Comment:
This API design is a bit weird. I would throw exception for this and provide another API called "isEmpty()" to check emptiness.